feat: Made View Individual Role Listings Dynamic and Added Viewing of Matched and Unmatched skills

christianmk04 opened the pull request:
### 📋 Overview (what is this PR about) 

Made View Individual Role Listings Dynamic, so now when a staff member clicks on a role listing from the view available role listings page, the details of the role with the corresponding ID should show.

For each of the role listings, the user should be able to see all the skills that they have, as well as the skills they have that are matched and all the skills they don't have (unmatched skills). They should be able to see the number and the percentage of matched skills for the role listing.

For the skills that they have, user can click on a skill chip to view the description of the skill as a modal popup.

### 🚀 What are the features/changes included?
URL to call the specific role listing from the backend is no longer hardcoded by dynamically taken from the id in the url of the page.

### 🧪 What testing have I done to make sure it is working?
Have manually went to click on several role listings on the View Available Role Listings page to see if the correct details are shown.

### 🧐 How should the reviewer go about testing it?
Reviewer can click on any role listings on the View Available Role Listings page and take a look if the correct details are shown

### ✅ Final check
- [ ] I have done the following:
  - `<git rebase main>` to replay your feature commits from the main branch
  - If there are any conflicts, resolve them then run:
    - `git add <resolved rebase conflict files>`
    - `<git rebase --continue>`
    - You may need to run `<git pull>` if your local feature branch is behind the remote feature branch
    - `<git push>`

### 🖼️ Make it visual ✨ (if possible, attach a screenshot of what the reviewer is supposed to see)

<img width="1238" alt="image" src="https://github.com/IS212-G9-T5/skill-based-role-portal/assets/111347751/0f92b545-ed18-4a32-9041-4eee52ac3e2a">
Role Listing for Role with ID 20
 
<img width="1239" alt="image" src="https://github.com/IS212-G9-T5/skill-based-role-portal/assets/111347751/ad8f4f17-90b0-4dc0-8c17-2fb94c8acec5">
Role Listing for Role with ID 156

### ✍️ Any additional notes
NIL

zhiweit committed:
feat: paginate role listings

zhiweit committed:
feat: get individual role listing

zhiweit committed:
chore: shift gitignore to backend

christianmk04 committed:
Merge branch 'T5-44-feature/paginate-role-listings' of https://github.com/IS212-G9-T5/skill-based-role-portal

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal

christianmk04 committed:
feat: Prototype Update Listings Page

created a prototype update listings page for testing

christianmk04 committed:
feat: Prototype Update Listings Page

created a prototype update listings page for testing

christianmk04 committed:
feat: Prototype Update Listings Page

created a prototype update listings page for testing

christianmk04 committed:
feat: Prototype Update Listings Page

created a prototype update listings page for testing

christianmk04 committed:
feat: Prototype Update Listings Page

created a prototype update listings page for testing

christianmk04 committed:
feat: Prototype Update Listings Page

created a prototype update listings page for testing

christianmk04 committed:
feat: Dynamic retrieval of View Individual Role Listings

dynamically retrieved id from the url of the view individual role listings page to retrieve the appropriate role listing

christianmk04 committed:
feat: Dynamic retrieval of View Individual Role Listings

dynamically retrieved id from the url of the view individual role listings page to retrieve the appropriate role listing

christianmk04 committed:
feat: Dynamic retrieval of View Individual Role Listings

dynamically retrieved id from the url of the view individual role listings page to retrieve the appropriate role listing

christianmk04 committed:
removed prototype updateRoleListings page

christianmk04 committed:
removed prototype updateRoleListings page

christianmk04 committed:
removed prototype updateRoleListings page

Jaylin0312 commented on frontend/src/pages/applicant/main.tsx:
Consider using `fetch("/api/...")` (either relative path or URL) instead of `axios` since we are not making any complex requests, and also to make the frontend code consistent (the rest of the components are using `Fetch`).

More about how to use `Fetch`: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
More about difference between `Fetch` and `Axios`: https://www.geeksforgeeks.org/difference-between-fetch-and-axios-js-for-making-http-requests/

christianmk04 replied:
It is already using fetch to call

zhiweit committed:
feat: search listing by role and filter by skills

zhiweit committed:
feat: search listing by role and filter by skills

zhiweit committed:
feat: search listing by role and filter by skills

zhiweit committed:
feat: return skills matched/unmatched for individual role listing

zhiweit committed:
feat: return skills matched/unmatched for individual role listing

zhiweit committed:
feat: return skills matched/unmatched for individual role listing

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal

christianmk04 committed:
Merge branch 'T5-81-feat-BE-search-role-listing-by-role-name' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

christianmk04 committed:
feat: able to show skills matched

christianmk04 committed:
feat: able to show skills matched

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal

christianmk04 committed:
Merge branch 'frontend/update-role-listings-page' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

christianmk04 committed:
feat: Added Skills Matching display

For an individual skills listing, able to see the skills that a user has that is matched to the role required and skills that are not matched, as well as the number and percentage of matched skills

christianmk04 committed:
feat: Added Skills Matching display

For an individual skills listing, able to see the skills that a user has that is matched to the role required and skills that are not matched, as well as the number and percentage of matched skills

christianmk04 committed:
feat: Changed the viewing of matched skills from Green/Red to Bold and Disabled

christianmk04 committed:
feat: Changed the viewing of matched skills from Green/Red to Bold and Disabled

christianmk04 committed:
feat: adding modal (MODAL BUG)

christianmk04 committed:
feat: adding modal (MODAL BUG)

zhiweit committed:
chore: migrate db schema

christianmk04 committed:
refactor: changed the name of main.tsx to IndividualRoleListing.tsx

christianmk04 committed:
Merge branch 'frontend/update-role-listings-page' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

christianmk04 committed:
feat: User Profile Page to View Skills

Created a User Profile page which allows the User to view the skills of the user as well as profile details

christianmk04 committed:
feat: added handling for negative path

added handling for negative path where if the user does not have any skills

zhiweit committed:
chore: update auth test cases for role listing endpoint

zhiweit committed:
Merge branch 'T5-81-feat-BE-search-role-listing-by-role-name' of https://github.com/IS212-G9-T5/skill-based-role-portal into T5-79-chore-update-test-case-auth-role-listing

christianmk04 committed:
refactor: moved skills profile user story pages and routes to another branch

zhiweit committed:
feat: include skill description in individual listing endpoint

zhiweit committed:
feat: update skills endpoint response body

Jaylin0312 committed:
chore: added comments

Jaylin0312 committed:
fix: login auth

Jaylin0312 committed:
fix: (FE) view and create role listings

Jaylin0312 committed:
refactor: HR keyword in allowed permission

christianmk04 committed:
Merge branch 'T5-79-chore-update-test-case-auth-role-listing' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

christianmk04 committed:
refactor: refactored code to accommodate for new BE body

christianmk04 committed:
Merge branch 'main' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

christianmk04 committed:
refactor: changed the parameter type in getRoleListingById in RoleListingAPI.ts

Also commented out the import axios in AllRoleListing.tsx as it was not being used an I could not build if i did not comment it out

Jaylin0312 commented on frontend/src/index.css:
Will suggest to use tailwindcss instead of having another css file since we are using tailwindcss alr :DD

Jaylin0312 commented on frontend/src/pages/applicant/AllRoleListing.tsx:
```diff
@@ -5,6 +5,7 @@ import { Link, useLocation } from "react-router-dom"
 import { getRoleListings } from "../../api/RoleListingAPI"
 import Navbar from "../../components/Navbar"
 import Role from "../../components/Role"
+// import axios from "axios"
```
Can just remove this instead of commenting

Jaylin0312 commented on frontend/src/types.d.ts:
Is this a new file? There should already be a global type file under `src/types/global.d.ts`

christianmk04 replied:
i think when i merged the last time from someone's branch, this file was missing so i created a new one but then when i rebased or merged this file came back, but yeah i think its the same

christianmk04 replied:
will remove this types.d.ts

christianmk04 committed:
Deleted node_modules from backend

christianmk04 committed:
refactor: changed css to tailwindcss

remove extra types.d.ts file and updated the types in global.d.ts

christianmk04 committed:
Update RoleListing.tsx

Jaylin0312 committed:
fix: removed yarn files

Jaylin0312 commented on frontend/src/pages/applicant/main.tsx:
```diff
@@ -2,22 +2,52 @@ import { useEffect, useState } from "react"
+    }
+  
+    fetchSkills();
+  }, []);
```
Can help to refactor by extracting this logic to api folder (same as other api calls)

Jaylin0312 replied:
Updated in latest commit

christianmk04 committed:
refactor: changed code for fetching user skills and individual role listing page

refactored fetchskills and moved it to RoleListingAPI.ts and changed the code for displaying the skills and their modal popups on the individual role listing page

christianmk04 committed:
Merge branch 'frontend/update-role-listings-page' of https://github.com/IS212-G9-T5/skill-based-role-portal into frontend/update-role-listings-page

Jaylin0312 commented:
Overall looks good to me, thanks for the nice work! Merging this into main
